src/ol/format/GML2.js

```diff
@@ -590,7 +590,7 @@ class GML2 extends GMLBase {
  * @type {Object<string, Object<string, import("../xml.js").Parser>>}
  * @private
  */
-GML2.prototype.GEOMETRY_FLAT_COORDINATES_PARSERS_ = {
+GML2.prototype.GEOMETRY_FLAT_COORDINATES_PARSERS = {
```

Comment:
Why does typescript complain ?
Shouldn't we remove things from prototype ?

Reply:
The TypeScript error is caused by using `this. GEOMETRY_FLAT_COORDINATES_PARSERS_` in `GMLBase`, which is why this needs to be protected. Creating it here creates one instance on the prototype, versus putting it in the constructor and recreating this for each instance. This is the same pattern used for `GMLBase.prototype.RING_PARSERS`.